pkg/shell/base_index.js

```diff
@@ -76,7 +74,8 @@ function Frames(index, setupIdleResetTimers) {
         frame.timer = null;
 
         try {
-            ready = $("body", frame.contentWindow.document).is(":visible");
+            if (frame.contentWindow.document && frame.contentWindow.document.body)
+                ready = frame.contentWindow.document.body.offsetWidth > 0 && frame.contentWindow.document.body.offsetHeight > 0;
```

Comment:
Is non-zero offset what jQuery defined as ":visible"? I.e. that's not the same as the `hidden` attribute or `display: none` style, but more like "finished loading"?

Reply:
[Since jquery 1.3.2](https://blog.jquery.com/2009/02/20/jquery-1-3-2-released/):

> an element is visible if its browser-reported offsetWidth or offsetHeight is greater than 0.

Reply:
And yes, this is until page shows nothing in the frame - for that duration we show spinner in the top row. Once page shows something (usually spinner), then we remove the spinner from top navigation and let frame itself handle loading.